src/comparator/field_comparator.py

```diff
@@ -12,17 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from google.api import resource_pb2
 from google.protobuf.descriptor_pb2 import FieldDescriptorProto
 from src.findings.finding_container import FindingContainer
 from src.findings.utils import FindingCategory
 
 
 class FieldComparator:
+    # global_resources: file-level resource definitions.
+    # local_resource: message-level resource definition.
+    # We need the resource database information to determine if the resource_reference
+    # annotation removal or change is breaking or not.
     def __init__(
-        self, field_original: FieldDescriptorProto, field_update: FieldDescriptorProto
+        self,
+        field_original: FieldDescriptorProto,
+        field_update: FieldDescriptorProto,
+        global_resources_original=None,
+        global_resources_update=None,
+        local_resource_original=None,
+        local_resource_update=None,
     ):
```

Comment:
This will be refactored to just take two objects, please see PR description for more details.